packages/core/src/utils/__tests__/security.spec.ts

```diff
@@ -0,0 +1,332 @@
+import { validateSecurity } from '../security';
+
+describe('validateSecurity', () => {
+  describe('when no security schemes are defined in a spec', () => {
+    it('passes the validation', () => {
+      expect(validateSecurity<any, any>({}, { security: [] })).toStrictEqual([]);
+    });
+  });
+
+  describe('when given an unknown security scheme type', () => {
+    it('fails with a message explaining the issue', () => {
+      expect(validateSecurity<any, any>({}, { security: [{}] })).toStrictEqual([
+        {
+          message: 'No handler for the security scheme found.',
```

Comment:
Hey, what does 'no handler found' mean? Is that what we're sending to the user?
WDYT about rephrasing to something like:

```suggestion
          message: 'Security scheme type is not defined.',
```

Or maybe it should silently pass if we provide invalid security scheme?

Reply:
Well, only some security schemes are handled (like Basic/Digest/Bearer). It would be impossible/hard to handle all types of authorization schemes (@philsturgeon mentioned this link before https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml). So, for example, if a user specifies in a spec:
```
components:
  securitySchemes:
    VapidAuth:
      type: http
      scheme: vapid
```
then security scheme (and its type) is defined, but we don't have a handler for it.

Reply:
So maybe "we currently do not support this type of security scheme"? Or something along those lines? Less cryptic imo.